### PR TITLE
Document how to bin/scratch launched items into Grafana

### PR DESCRIPTION
**Pull request by quodlibetor**
The existing docs had examples that were correct, but there was no
documentation for exactly what was required, making it difficult to be sure
that everything worked if you were building something from, ahem, scratch.

### Checklist

- [n/a] This PR has adequate test coverage / QA involvement has been duly considered.
- [n/a] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: doc/developer/scratch.md [hunk 1]
```diff
@@ -124,5 +126,87 @@ bin/scratch destroy --all-mine
 Pass `--dry-run` if you want to see what instances `bin/scratch destroy` would
 destroy without actually destroyin them.
 
+### Grafana integration
+
+The `materialized` process always exposes metrics at via its primary port's http server on the
```
Comment (benesch) [inline doc/developer/scratch.md:131]:
```suggestion
The `materialized` process always exposes metrics at its primary port's HTTP server on the
```

File: doc/developer/scratch.md [hunk 2]
```diff
@@ -124,5 +126,87 @@ bin/scratch destroy --all-mine
...
+prometheus-standard `/metrics` path, but the deployed server needs to be configured correctly for
```
Comment (benesch) [inline doc/developer/scratch.md:132]:
```suggestion
prometheus-standard `/metrics` path, but the scratch instance needs to be configured correctly for
```

(Wasn't clear to me whether "server" referred to the just-mentioned `materialized` process or the scratch instance; I think this is what you meant so I figured I'd make it explicit.)

File: doc/developer/scratch.md [hunk 3]
```diff
@@ -124,5 +126,87 @@ bin/scratch destroy --all-mine
...
+our Prometheus server to actually scrape the metrics and thereby expose them to Grafana. You
+
+The tl;dr is that you must configure mzcompose with `--preserve-ports` and the EC2 instance with
+the `"scrape_benchmark_numbers": "true"` tag. Thus a bare-minimum grafana-integrate config looks
```
Comment (benesch) [inline doc/developer/scratch.md:136]:
```suggestion
the `"scrape_benchmark_numbers": "true"` tag. Thus a bare-minimum Grafana-integrated config looks
```

File: doc/developer/scratch.md [hunk 4]
```diff
@@ -124,5 +126,87 @@ bin/scratch destroy --all-mine
...
+```
+
+Read on for more details and some other items that can be configured.
+
+#### Prometheus config
+
+There are three tags that configure our Prometheus' integration, and if you are running inside
+mzcompose it must also be configured correctly.
+
+The three Prometheus tags are:
```
Comment (benesch) [inline doc/developer/scratch.md:156]:
```suggestion
There are three tags on the EC2 instance that configure our Prometheus integration:
```

(I think it makes more sense not talk about mzcompose here; you've already talked about it in general in the overview section above and you've got a dedicated section for it below!)

File: doc/developer/scratch.md [hunk 5]
```diff
@@ -124,5 +126,87 @@ bin/scratch destroy --all-mine
...
+    // .. snip general config ..
+    "tags": {
+        "scrape_benchmark_numbers": "true",
+        "purpose": "bwm-debugging",
+        "test": "billing"
+    }
+}
+```
+
+#### MzCompose config
```
Comment (benesch) [inline doc/developer/scratch.md:198]:
```suggestion
#### mzcompose config
```

File: doc/developer/scratch.md [hunk 6]
```diff
@@ -124,5 +126,87 @@ bin/scratch destroy --all-mine
...
+
+Prometheus only looks for materialize metrics on port 6875. The canonical way to ensure that
+Materialize is available on port 6875 on a host is to pass the `--preserve-ports` argument to
+mzcompose.
```
Comment (benesch) [inline doc/developer/scratch.md:202]:
```suggestion
Prometheus only looks for Materialize metrics on port 6875. The canonical way to ensure that
Materialize is available on port 6875 on a host is to pass the `--preserve-ports` argument to
mzcompose. (Without this flag, mzcompose chooses a random host port for Materialize, which
will be unknown to Prometheus.)
```

(Just added a small note about what goes wrong if you don't specify `--preserve-ports`.)

**Review by benesch — APPROVED** [review]
Thanks for doing this! I wonder if we should make `scrape_benchmark_numbers` the default...

File: doc/developer/scratch.md [hunk 7]
```diff
@@ -124,5 +126,87 @@ bin/scratch destroy --all-mine
...
+There are three tags that configure our Prometheus' integration, and if you are running inside
+mzcompose it must also be configured correctly.
+
+The three Prometheus tags are:
+
+* `scrape_benchmark_numbers`: must be set to exactly the string `"true"` in order for Prometheus
+  to observe the instance.
+* `purpose`: is used as a filter in the Grafana UI. You can use this to group all your instances
+  (e.g. set it to `myname-debugging`) or set it to `load-test` or `benchmark`.
+* `test` is both used by scratch to determine which test to execute, and displays as an additional
```
Comment (ruchirK) [inline doc/developer/scratch.md:162]: wait so just to double check - if the value of test doesn't match whats actually run in the launch script will that be a problem? I couldn't tell if the `test` field was purely informational and it seems like no?

**Review by ruchirK — APPROVED** [review]
LGTM! I also think scrape_benchmark_numbers true is a nice default unless theres some important reason we don't already do so

File: doc/developer/scratch.md (hunk 7 above)
Comment (quodlibetor) [reply doc/developer/scratch.md:159]: Test is used to determine what test to automatically run by the scratch script. If it is invalid then nothing will be run automatically.

Comment (benesch) [reply doc/developer/scratch.md:159]: > Test is used to determine what test to automatically run by the scratch script.

Hrm, are you sure? Where are you seeing that? Am I just missing it?

File: doc/developer/scratch.md [hunk 8]
```diff
@@ -124,5 +126,87 @@ bin/scratch destroy --all-mine
 Pass `--dry-run` if you want to see what instances `bin/scratch destroy` would
 destroy without actually destroyin them.
 
+### Grafana integration
+
+The `materialized` process always exposes metrics at via its primary port's http server on the
+prometheus-standard `/metrics` path, but the deployed server needs to be configured correctly for
+our Prometheus server to actually scrape the metrics and thereby expose them to Grafana. You
```
[the remark this answers is not in the document]
Comment (ruchirK) [reply doc/developer/scratch.md:133]: very gentle bump to remove the dangling "You"

File: doc/developer/scratch.md (hunk 7 above)
Comment (quodlibetor) [reply doc/developer/scratch.md:159]: Ah, no I must have been remembering the way it used to work, or just wrong.

**Pull request merged**